pkg/controllers/registryclient/client_controller.go

```diff
@@ -55,6 +58,7 @@ func newRegistryClient(
 		reDial:              make(chan struct{}),
 		stopCh:              make(chan struct{}),
 		namespace:           namespace,
+		allowedDomains:      cloneStringSlice(remoteRegistry.Spec.AllowedDomains),
```

Comment:
The worry is that with the way the test is set up, making changes to the RemoteRegistry struct in the test setup may unintentionally change the struct referenced inside this controller.

Reply:
I've removed this code since it appears the worry is greater than just the allowedDomains field. It has been replaced with a deep copy on the whole RemoteRegistry struct due to data races.